### ISSUE TITLE
no-multi-spaces ignores multiple spaces before inline comments

### ISSUE THREAD

**Issue by minj** [eslint/eslint#7693]
**Tell us about your environment**

* **ESLint Version:** v3.10.2
* **Node Version:** v7.2.0
* **npm Version:** 4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue.**
```bash
$ eslint --no-eslintrc foo.js
```
```js
/*eslint no-multi-spaces: "error"*/
console.log('hello',  'world');  // comment
```
**What did you expect to happen?**
Two problems should be reported (`'world'`, `// comment`)
**What actually happened? Please include the actual, raw output from ESLint.**
```

foo.js
  2:23  error  Multiple spaces found before ''world''  no-multi-spaces

✖ 1 problem (1 error, 0 warnings)
```

**Comment by kaicataldo** [comment]
Confirmed. It does seem like a bug that this rule doesn't check inline comments.

**Comment by platinumazure** [comment]
Another test case (may already be covered): `console.log(/* comment followed by multiple spaces */         foo);`

**Comment by not-an-aardvark** [comment]
I'm fine with changing this, but keep in mind that some people might like this kind of alignment:

```js
foo();    // do the first thing
foobar(); // also do the second thing
```

**Comment by platinumazure** [comment]
Maybe we could make this an enhancement via new option?

**Comment by kaicataldo** [comment]
@platinumazure Do you mean adding checking comments at all via new option, or inline comments at the end of the line, as @not-an-aardvark mentioned? I'm fine with that, though I do think the following should always warn:

```js
console.log('hello',     /* comment */ 'world');
console.log('hello', /* comment */     'world');
```

**Comment by kaicataldo** [comment]
That being said, I don't think an option is necessary. As a user, I would expect these cases to be flagged.

**Comment by platinumazure** [comment]
I was thinking end of line checks as being relaxable. I agree that comments
in the middle of lines should be checked by default. I also have no problem
with just fixing this as a semver-minor fix, since users can disable the
rule or use `// eslint-disable-next-line` as needed.

**Comment by kaicataldo** [comment]
@platinumazure I vote for semver-minor fix and, if we see requests for an option to relax comments at the end of lines, implement the option then

**Comment by kaicataldo** [comment]
As an update: looks like consensus is to put this behind an option. Thoughts on an option name? `exceptEOLComments`?

I'm going to be less active over the next week, but plan to get back to this in the new year.

**Comment by minj** [comment]
Seems reasonable assuming `false` is the default

**Comment by not-an-aardvark** [comment]
I would prefer an option name like `ignoreEOLComments` rather than `exceptEOLComments`.

@minj I think the default behavior would be the same as the current behavior, to avoid breaking peoples' builds.

**Comment by minj** [comment]
That's what I meant. The option name should reflect non-default behavior so that `false` is the default.

either `ignoreEOLComments=false` or `exceptEOLComments=false` by default.

**Comment by minj** [comment]
err, I meant `includeEOLComments=false`

**Comment by kaicataldo** [comment]
This is currently blocked by the outcome of this discussion: https://github.com/eslint/eslint/issues/7810

**Comment by kaicataldo** [comment]
PR opened as I don't think we need to block this any more (see #7810 for more details)
